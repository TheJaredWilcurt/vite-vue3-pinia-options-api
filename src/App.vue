<template>
  <header>
    <nav>
      <RouterLink :to="{ name: 'welcome' }">
        Welcome
      </RouterLink>
      <RouterLink :to="{ name: 'pinia' }">
        Pinia Demo
      </RouterLink>
      <RouterLink :to="{ name: 'resources' }">
        Resources
      </RouterLink>
    </nav>
    <img
      alt="Vue logo"
      class="logo"
      src="./assets/logo.svg"
    />

  </header>

  <main>
    <RouterView />
  </main>
</template>

<script>
import { APP_NAME } from '@/helpers/constants.js';

export default {
  name: 'App',
  constants: {
    APP_NAME
  }
};
</script>

<style scoped>
header {
  line-height: 1.5;
}

nav {
  margin-bottom: 2rem;
  text-align: center;
}
nav a {
  background: #FFF3;
  border-radius: 0.5rem;
  margin: 2rem;
  padding: 0.5rem 1rem;
}
nav a:hover {
  background: #FFF5;
  color: #FFF;
}
.router-link-active {
  background: #FFF7;
  color: #FFF;
}
.router-link-active:hover {
  background: #FFF8;
}

.logo {
  display: block;
  width: 125px;
  height: 125px;
  margin: 0px auto 2rem;
}

</style>
