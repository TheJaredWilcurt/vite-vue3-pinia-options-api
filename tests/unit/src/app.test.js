import App from '@/App.vue';

import testHelpers from '@@/unit/testHelpers.js';

describe('App', () => {
  const setupWrapper = async () => {
    const wrapper = await testHelpers.shallowMount(App);
    return wrapper;
  };

  test('Renders correctly', async () => {
    const wrapper = await setupWrapper();

    expect(wrapper)
      .toMatchSnapshot();
  });
});
