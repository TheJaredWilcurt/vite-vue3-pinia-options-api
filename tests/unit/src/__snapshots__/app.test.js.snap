// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders correctly 1`] = `
<header>
  <nav>
    <router-link-stub
      ariacurrentvalue="page"
      custom="false"
      replace="false"
      to="{ name: 'welcome' }"
      viewtransition="false"
    ></router-link-stub>
    <router-link-stub
      ariacurrentvalue="page"
      custom="false"
      replace="false"
      to="{ name: 'pinia' }"
      viewtransition="false"
    ></router-link-stub>
    <router-link-stub
      ariacurrentvalue="page"
      custom="false"
      replace="false"
      to="{ name: 'resources' }"
      viewtransition="false"
    ></router-link-stub>
  </nav>
  <img
    alt="Vue logo"
    class="logo"
    src="/src/assets/logo.svg"
  />
</header>
<main>
  <router-view-stub name="default"></router-view-stub>
</main>
`;
